docs/IPXE.md

```diff
@@ -0,0 +1,24 @@
+# IPXE
+
+In Boots we build custom IPXE binaries for different architectures.
```

Comment:
worth a call out to `make gen` and ../ipxev.mk?

Reply:
`make gen` doesn't actually do any ipxe related things anymore now that we've moved to builtin `go:embed`.
Would you mind if I came back around to add details about `ipxev.mk`?

Reply:
oh darn I missed that then. How do we build ipxe binaries now?

Reply:
`make ipxe` and `make build_all_ipxe` handle all that now.

Reply:
the Go code embedding now happens at `go build` time.

Reply:
> Would you mind if I came back around to add details about `ipxev.mk`?

:+1:

Reply:
> `make ipxe` and `make build_all_ipxe` handle all that now.

Why get rid of iPXE building from `make gen`, I thought of it like a generic "generate all the deps" target.

Reply:
that's fair. I didn't think of ipxe builds as being generated so I put it elsewhere. I like them on their own too for maintainability and composability.